airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -133,11 +128,50 @@ def _get_request_options(
         :return:
         """
         requester_mapping = requester_method(self.state, stream_slice, next_page_token)
-        paginator_mapping = paginator_method()
-        keys_intersection = set(requester_mapping.keys()) & set(paginator_mapping.keys())
-        if keys_intersection:
-            raise ValueError(f"Duplicate keys found: {keys_intersection}")
-        return {**requester_mapping, **paginator_mapping}
+        paginator_mapping = paginator_method(self.state, stream_slice, next_page_token)
+        stream_slicer_mapping = stream_slicer_method(stream_slice)
+        requester_paginator_intersection = set(requester_mapping.keys()) & set(paginator_mapping.keys())
```

Comment:
why are we doing it this way instead of just intersecting all three sets at once?

Reply:
we can't intersect all 3 at once because it'll only find elements present in all sets
```
>>> A = set(["A"])
>>> B = set(["A"])
>>> C = set(["C"])
>>> (A & B & C)
set()
```

but we can simplify this to
```
>>> (A & B | A & C | B & C)
{'A'}
```

Reply:
derp yes of course set theory